api-resources.md

```diff
@@ -0,0 +1,414 @@
+## Resources
+
+DAP is structured as an HTTP-based application with the following resources:
+
+- (TODO) Tasks
+- HPKE configurations
+- Reports
+- Aggregation jobs (and lists thereof)
+- Aggregate shares (and lists thereof)
+- Collections (and lists thereof)
+
+A resource's path is resolved relative to a server's base URL to construct a resource URI. Deployments may host resource paths arbitrarily deep relative to their domain. Paths are generally structured as `/resource-type/resource-id`. Anywhere `resource-id` (e.g., `task-id` or `report-id`) occurs in a URI is to be understood as the URL-safe, unpadded base64 representation of the resource's identifier, which itself is usually 16 random bytes (some are 32, but https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/349 will normalize them to 16).
+
+Some resources are owned by another resource, and so their URI will contain two identifiers. For instance, a report belongs to a task, so the URI is `[base]/tasks/task-id/reports/report-id`.
+
+Generally, if a resource supports the `GET` method, it also supports `HEAD`. The response to `HEAD` is exactly the same as `GET`, but the response contains no body.
+
+If an HTTP method is listed in a resource's required HTTP methods, implementations MUST provide responses as indicated. Otherwise, implementations MAY implement methods on resources however they wish, including returning an error like HTTP 405 Method Not Allowed.
+
+We use HTTP `PUT` requests for idempotent creation of resources. Server responses to `PUT` requests follow [RFC 9110 section 9.3.4](https://httpwg.org/specs/rfc9110.html#rfc.section.9.3.4).
+
+Finally, a quick reminder of some HTTP semantics: `GET` and `PUT` are idempotent (i.e., the same request may be made multiple times without changing state in the server more than once, and the response will be the same each time). `POST` is not idempotent. `GET` requests may not have bodies.
+
+We care a lot about idempotence because we need to account for cases where a client makes a request but never sees the response, and thus needs a way to recover from this state of uncertainty.
+
+### Tasks
+
+TODO: do we want a task resource that we can GET and PUT in the protocol? Does this collide with https://wangshan.github.io/draft-wang-ppm-dap-taskprov/draft-wang-ppm-dap-taskprov.html or https://github.com/divergentdave/draft-dcook-ppm-dap-interop-test-design?
```

Comment:
I think it can work with taskprov, which specifies how to provision a task in-band, it could end up calling the `PUT tasks/<task-id>` API. But so far the out-of-band task provision is not specified, so we may want to flesh that out (if we want to flesh that out) first.

Reply:
@tgeoghegan what I meant was after the task has been provisioned following task_prov extension, the leader/helper can call a _PUT to /tasks/task-id_ to store the task object. The /tasks endpoint should not be exposed to client.